pandas/io/formats/style.py

```diff
@@ -2211,21 +2264,60 @@ def _default_formatter(x: Any, precision: int) -> Any:
         Input variable to be formatted
     precision : Int
         Floating point precision used if ``x`` is float or complex.
+    thousands : bool
+        Whether to group digits with thousands separated with ",".
 
     Returns
     -------
     value : Any
-        Matches input type, or string if input is float or complex.
+        Matches input type, or string if input is float or complex or int with sep.
     """
     if isinstance(x, (float, complex)):
+        if thousands:
+            return f"{x:,.{precision}f}"
         return f"{x:.{precision}f}"
+    elif isinstance(x, int) and thousands:
+        return f"{x:,.0f}"
     return x
 
 
+def _maybe_wrap_deci_thou(
+    formatter: Callable, decimal: str = ".", thousands: Optional[str] = None
+) -> Callable:
+    """
+    Takes a string formatting function and wraps logic to deal with thousands and
+    decimal parameters, in the case that they are non-standard and that the input
+    is a (float, complex).
+    """
+
+    def wrapper(x):
```

Comment:
can we share code with FloatArrayFormatter in pandas/io/format.py ?

may need to refactor to do that.

Reply:
taking a quick look:

 - there doesn't seem to be an inherent option for `thousands`, just `decimal` replacement in that code.
 - the FloatArrayFormatter is vectorised but `Styler` does `display_funcs` element-wise (which it can't avoid for processing individual cell styles). I guess this makes it very difficult to refactor across module (unless the fomat_class was called element-wise)
 - other parameters such as ``escape`` and ``subset`` would have to be wrapped around the format.py code, so it might end up equally complicated anyway?

will have a think over next day.

Reply:
@jreback I think the refactor is too complicated at this stage since:

 - FloatArrayFormatter doesnt have the `thousands` argument which is main part of this PR.
 - the above problem about column vectorisation versus element-wise (and the `styler` `subset` argument)
 - the other arguments finding it difficult to exactly replicate existing behaviour.